api/envoy/config/filter/http/jwt_authn/v2alpha/config.proto

```diff
@@ -183,44 +207,209 @@ message JwtHeader {
+//
+// Example 2:
+//
+// .. code-block:: yaml
+//
+//    - match:
+//         prefix: "/"
+//      requires: { provider_name: "provider-A" }
+//
+// In above example, all requests matched the path prefix require jwt authentication
+// from "provider-A".
+message RequirementRule {
+  // The route matching parameter. Only when the match is satisfied, the "requires" field will
+  // apply.
   //
-  // Examples: bypass all CORS options requests
+  // For example: following match will match all requests.
   //
   // .. code-block:: yaml
   //
-  //   bypass:
-  //   - headers:
-  //     - name: :method
-  //       value: OPTIONS
-  //   - headers:
-  //     - name: :path
-  //       regex_match: /.*
+  //    match:
+  //       prefix: "/"
   //
-  // Examples: bypass /healthz check
+  envoy.api.v2.route.RouteMatch match = 1 [(validate.rules).message.required = true];
+
+  // Specify a Jwt Requirement. Please detail comment in message JwtRequirement.
+  JwtRequirement requires = 2;
+}
+
+// This is the Envoy HTTP filter config for JWT authentication.
+//
+// For example:
+//
+// .. code-block:: yaml
+//
+//   providers:
+//      - name: provider1
+//        issuer: issuer1
+//        audiences:
+//        - audience1
+//        - audience2
+//        remote_jwks:
+//          http_uri:
+//            uri: https://example.com/.well-known/jwks.json
+//            cluster: example_jwks_cluster
+//      - name: provider2
+//        issuer: issuer2
+//        local_jwks:
+//          inline_string: jwks_string
+//
+//   rules:
+//      # Not jwt verification is required for /health path
+//      - match:
+//          prefix: "/health"
+//
+//      # Jwt verification for provider1 is required for path prefixed with "prefix"
+//      - match:
+//          prefix: "/prefix"
+//        requires:
+//          provider_name: "provider1"
+//
+//      # Jwt verification for either provider1 or provider2 is required for all other requests.
+//      - match:
+//          prefix: "/"
+//        requires:
+//          requires_any:
+//            requirements:
+//              - provider_name: "provider1"
+//              - provider_name: "provider2"
+//
+//// [#not-implemented-hide:]
+message JwtAuthentication {
+  // List of JWT providers. For example:
   //
   // .. code-block:: yaml
   //
-  //   bypass:
-  //   - headers:
-  //     - name: :method
-  //       value: GET
-  //   - headers:
-  //     - name: :path
-  //       exact_match: /healthz
+  //   providers:
+  //      - name: provider1
+  //        issuer: issuer1
+  //        audiences:
+  //        - audience1
+  //        - audience2
+  //        remote_jwks:
+  //          http_uri:
+  //            uri: https://example.com/.well-known/jwks.json
+  //            cluster: example_jwks_cluster
+  //      - name: provider2
+  //        issuer: provider2
+  //        local_jwks:
+  //          inline_string: jwks_string
+  //
+  repeated JwtProvider providers = 1;
+
+  reserved 2;
```

Comment:
Forgive me if this is obvious, but why is this here? Is there something already in mind for this id, and is there a reason it shouldnt go at the end?

Reply:
Is binary compatibility necessary? Seems that these protos are substantially different from what was in istio, and they shouldn't be expected to interoperate. Maybe there's a lot going on over there that I haven't seen though.

Reply:
We can clean up those when this proto become v2 or v2beta, keeping this for now if istio can manage the compatibility.

Reply:
I think we don't need to support binary compatible.  Filter name will be different.  Istio Pilot need to change filter name anyway.